src/net/connection/connection_pool.rs

```diff
@@ -1,53 +1,31 @@
-use super::VirtualConnection;
-use error::{NetworkError, NetworkResult};
+use super::{VirtualConnection, ConnectionsCollection, Connection};
+use error::{NetworkResult, NetworkErrorKind, NetworkError};
 use events::Event;
 use net::NetworkConfig;
-
-use log::info;
+use log::{error, info};
 use std::collections::HashMap;
-use std::error::Error;
 use std::net::SocketAddr;
 use std::sync::mpsc::Sender;
 use std::sync::{Arc, RwLock};
-use std::thread;
 use std::time::Duration;
+use std::error::Error;
 
-pub type Connection = Arc<RwLock<VirtualConnection>>;
-pub type Connections = HashMap<SocketAddr, Connection>;
-pub type ConnectionsCollection = Arc<RwLock<Connections>>;
-
-// Default time between checks of all clients for timeouts in seconds
-const TIMEOUT_POLL_INTERVAL: u64 = 1;
-const TIMEOUT_DEFAULT: u64 = 10;
 /// This is a pool of virtual connections (connected) over UDP.
 pub struct ConnectionPool {
-    /// All the connections we are aware of
     connections: ConnectionsCollection,
-    /// This is how long a connection is allowed to go unheard from before we mark it as disconnected
-    #[allow(dead_code)]
-    sleepy_time: Duration,
-    /// This is how frequently we check the connections for activity
-    poll_interval: Duration,
-    /// The network config in use by this pool
-    config: Arc<NetworkConfig>,
+    config: Arc<NetworkConfig>
 }
 
 impl ConnectionPool {
-    /// Creates and returns a new ConnectionPool
     pub fn new(config: &Arc<NetworkConfig>) -> ConnectionPool {
-        let sleepy_time = Duration::from_secs(TIMEOUT_DEFAULT);
-        let poll_interval = Duration::from_secs(TIMEOUT_POLL_INTERVAL);
-
         ConnectionPool {
             connections: Arc::new(RwLock::new(HashMap::new())),
-            sleepy_time,
-            poll_interval,
-            config: config.clone(),
+            config: config.clone()
         }
     }
 
-    /// Insert connection if it does not exists.
-    pub fn get_connection_or_insert(&mut self, addr: &SocketAddr) -> NetworkResult<Connection> {
+    /// Try getting connection by address if the connection does not exists it will be inserted.
+    pub fn get_connection_or_insert(&self, addr: &SocketAddr) -> NetworkResult<Connection> {
         let mut lock = self
             .connections
             .write()
```

Comment:
I thought we fixed this? If the connection exists we will acquire a write lock but we should only be acquiring the read lock?

Reply:
Might me wrong but for inserting we need write

Reply:
Inserting a connection requires a write lock, yes

Reply:
fine we can address this later.

Reply:
You could get a read lock and check if it exists, and then do a write lock only if an insert is needed. But we can make an issue for that.

Reply:
#102